indexd_test_utils2/__init__.py

```diff
@@ -0,0 +1,296 @@
+    truncate_tables(driver, alias_base)
+    driver.dispose()
+
+
+@pytest.fixture(scope="session")
+def auth_driver(pg_url):
+    driver = SQLAlchemyAuthDriver(pg_url)
+    yield driver
+    driver.dispose()
+
+
+@pytest.fixture
+def indexd_admin_user(auth_driver):
+    username = password = "admin"
+    auth_driver.add(username, password)
+    yield username, password
+    auth_driver.delete("admin")
+
+
+@pytest.fixture
+def index_driver_no_migrate(pg_url):
+    """
+    This fixture is designed for testing migration scripts and can be used for
+    any other situation where a migration is not desired on instantiation.
+    """
+    driver = SQLAlchemyIndexDriver(pg_url, auto_migrate=False)
+    yield driver
+    truncate_tables(driver, index_base)
+    driver.dispose()
+
+
+@pytest.fixture
+def alias_driver_no_migrate(pg_url):
+    """
+    This fixture is designed for testing migration scripts and can be used for
+    any other situation where a migration is not desired on instantiation.
+    """
+    driver = SQLAlchemyAliasDriver(pg_url, auto_migrate=False)
+    yield driver
+    truncate_tables(driver, alias_base)
+    driver.dispose()
+
+
+@pytest.fixture
+def create_indexd_tables(index_driver, alias_driver, auth_driver):
+    """Make sure the tables are created but don't operate on them directly.
+    Also set up the password to be accessed by the client tests.
+    Migration not required as tables will be created with most recent models
+    """
+    pass
+
+
+@pytest.fixture
+def create_indexd_tables_no_migrate(
+        index_driver_no_migrate, alias_driver_no_migrate, auth_driver):
+    """Make sure the tables are created but don't operate on them directly.
+
+    There is no migration required for the SQLAlchemyAuthDriver.
+    Also set up the password to be accessed by the client tests.
+    """
+    pass
+
+
+@pytest.fixture
+def indexd_client(indexd_server, create_indexd_tables, indexd_admin_user):
+    """Create the tables and add an auth user"""
+    return IndexClient(indexd_server.baseurl, auth=(indexd_admin_user[0], indexd_admin_user[1]))
+
+
+@pytest.fixture(scope='session')
+def indexd_server(pg_url):
+    """
+    Starts the indexd server, and cleans up its mess.
+    Most tests will use the client which stems from this
+    server fixture.
+
+    Runs once per test session.
+    """
+    app = flask.Flask("indexd")
+    settings = indexd_settings.get_settings(pg_url)
+    app_init(app, settings)
+
+    hostname = 'localhost'
+
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    debug = False
+
+    port = random.randint(8000, 9000)
+    while sock.connect_ex((hostname, port)) == 0:
+        port = random.randint(8000, 9000)
+
+    t = threading.Thread(target=app.run, kwargs={'host': hostname, 'port': port, 'debug': debug})
+    t.setDaemon(True)
+    t.start()
+
+    wait_for_indexd_alive(port)
+    yield MockServer(port=port)
+
+
+def wait_for_indexd_alive(port):
+    url = f'http://localhost:{port}'
```

Comment:
The localhost should be pulled out, maybe to a method? This would be for any references to 'localhost'